### PR TITLE
[#399] [Bug] Fastlane Swift - Fix Cannot load such file -- ./fastlane/Constants/Constants

### PR DESCRIPTION
**Pull request by ducbm051291**
- #399 

## What happened

- Danger cannot load the file ./fastlane/Constants/Constants due to an incorrect path. 
 
## Insight

- Update defining constants in Danger file
- Update Automatic pull request review workflow using the new swift lane.
- Update Automatic pull request review workflow using macOS-12 instead of the latest. Ref: https://github.com/actions/runner-images/issues/6384

Note: We may update DangerSwift in the future for consistent with FastlaneSwift

## Proof Of Work

Screenshot 2022-12-20 at 20 59 18

### CODE REVIEW

File: fastlane/Constants/Constants.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+class Constants
```
Comment (suho) [inline fastlane/Constants/Constants.rb:3]: Please don't create this file, you need to define the constants directly in the Dangerfile

**Review by suho — CHANGES_REQUESTED** [review]

File: fastlane/Constants/Constants.rb (hunk 1 above)
Comment (ducbm051291) [reply fastlane/Constants/Constants.rb:3]: But how do we know the application's name before the user run make.sh command?

Comment (ducbm051291) [reply fastlane/Constants/Constants.rb]: Updated in ba738a1

File: Dangerfile [hunk 2]
```diff
@@ -23,7 +19,10 @@ swiftlint.lint_files(
   additional_swiftlint_args: '--strict'
 )
 
-xcresultPath = "#{Constants.TEST_OUTPUT_DIRECTORY_PATH}/#{Constants.TESTS_SCHEME}.xcresult"
+xcresultPath = "./fastlane/test_output/{PROJECT_NAME} Staging.xcresult"
+scheme = "{PROJECT_NAME} Staging"
+workspace = "./{PROJECT_NAME}.xcworkspace"
+outputDirectory = "./fastlane/xcov_output"
```
Comment (suho) [inline Dangerfile:25]: How about better naming for this?
```suggestion
xcovOutputDirectory = "./fastlane/xcov_output"
```

File: make.sh [hunk 3]
```diff
@@ -189,12 +189,6 @@ rm -rf .github/workflows/test_install_script.yml
 rm -f .git/index
 git reset
 
-if [[ -z "${CI}" ]]; then
```
Comment (suho) [inline make.sh:192]: Why you removed this?

Comment (ducbm051291) [reply make.sh:192]: Cause we don't use deliverable_setup.sh anymore, so I removed this. I will restore the rm part

Comment (suho) [reply make.sh:192]: I think we still need to use it, but with Swift Fastlane, please revert the changes

Comment (ducbm051291) [reply make.sh:192]: Done

Comment (suho) [reply make.sh:192]: @ducbm051291 Please revert all removed changes:
- make.sh
- set_up_test_testflight.sh
- deliverable_setup.sh

If the workflow does not work, please create a new task for that

Comment (ducbm051291) [reply make.sh]: Got it ;)

**Review by suho — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]

**Pull request merged**